pkg/splunk/enterprise/configuration.go

```diff
@@ -472,15 +474,32 @@ func getSplunkStatefulSet(client splcommon.ControllerClient, cr splcommon.MetaOb
 	return statefulSet, nil
 }
 
+// getAppListingConfigMap returns the App listing configMap, if it exists and applicable for that instanceType
+func getAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject, instanceType InstanceType) (*corev1.ConfigMap, bool) {
+
+	// ToDo: sgontla: Exclude MC, once it's own CR is available
+	if instanceType != SplunkIndexer && instanceType != SplunkSearchHead && instanceType != SplunkMonitoringConsole {
+		appListingConfigMapName := GetSplunkAppsConfigMapName(cr.GetName(), cr.GetObjectKind().GroupVersionKind().Kind)
+		return getConfigMap(client, cr, appListingConfigMapName)
+	}
+
+	return nil, false
+}
+
 // getSmartstoreConfigMap returns the smartstore configMap, if it exists and applicable for that instanceType
 func getSmartstoreConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject, instanceType InstanceType) (*corev1.ConfigMap, bool) {
-	var smartStoreConfigMapName string
 	if instanceType == SplunkStandalone || instanceType == SplunkClusterMaster {
-		smartStoreConfigMapName = GetSplunkSmartstoreConfigMapName(cr.GetName(), cr.GetObjectKind().GroupVersionKind().Kind)
+		smartStoreConfigMapName := GetSplunkSmartstoreConfigMapName(cr.GetName(), cr.GetObjectKind().GroupVersionKind().Kind)
+		return getConfigMap(client, cr, smartStoreConfigMapName)
 	}
 
-	if smartStoreConfigMapName != "" {
-		namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: smartStoreConfigMapName}
+	return nil, false
+}
+
+// getConfigMap returns configMap for a given name
+func getConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject, configMapName string) (*corev1.ConfigMap, bool) {
```

Comment:
This can be confused with GetConfigMap() API. What we are doing here is retrieving the configmap for the specific instanceType and checking if that needs to be mounted? OR we need to use the configmap Revision as the trigger?

Reply:
It started as fat logic, but now, its almost the same as GetConfgMap. I will get rid of it completely.